mars/services/storage/core.py

```diff
@@ -152,27 +176,51 @@ def put(self,
         info = InternalDataInfo(data_info, object_info)
         self._data_key_to_info[(session_id, data_key)].append(info)
         self._data_info_list[data_info.level][(session_id, data_key)] = object_info
+        if object_info is not None:
+            self._spill_strategy[data_info.level].record_put(
+                (session_id, data_key), data_info.store_size)
         if isinstance(data_key, tuple):
             self._main_key_to_sub_keys[(session_id, data_key[0])].update([data_key])
 
-    def get_infos(self,
-                  session_id: str,
-                  data_key: str) -> List[DataInfo]:
-        if (session_id, data_key) not in self._data_key_to_info:
-            if (session_id, data_key) in self._main_key_to_sub_keys:
-                infos = []
-                for sub_key in self._main_key_to_sub_keys[(session_id, data_key)]:
-                    infos.extend([info.data_info for info in
-                                  self._data_key_to_info.get((session_id, sub_key))])
-                return infos
-            raise DataNotExist(f'Data key {session_id, data_key} not exists.')
-        return [info.data_info for info in
-                self._data_key_to_info.get((session_id, data_key))]
-
-    def get_info(self,
-                 session_id: str,
-                 data_key: str,
-                 error: str = 'raise') -> Union[DataInfo, None]:
+    def _get_data_infos(self,
+                        session_id: str,
+                        data_key: str,
+                        error: str) -> Union[List[DataInfo], None]:
+        try:
+            if (session_id, data_key) not in self._data_key_to_info:
+                if (session_id, data_key) in self._main_key_to_sub_keys:
+                    infos = []
+                    for sub_key in self._main_key_to_sub_keys[(session_id, data_key)]:
+                        infos.extend([info.data_info for info in
+                                      self._data_key_to_info.get((session_id, sub_key))])
+                    return infos
+                raise DataNotExist(f'Data key {session_id, data_key} not exists.')
+            return [info.data_info for info in
+                    self._data_key_to_info.get((session_id, data_key))]
+        except DataNotExist:
+            if error == 'raise':
+                raise
+            else:
+                return
+
+    @extensible
+    def get_data_infos(self,
+                       session_id: str,
+                       data_key: str,
+                       error: str = 'raise') -> List[DataInfo]:
+        return self._get_data_infos(session_id, data_key, error)
+
+    @get_data_infos.batch
```

Comment:
Batch method not needed for sync methods.

Reply:
Removed